### PR TITLE
Migrate `beacon` from `ouroboros-consensus`.

### LINKED ISSUE #290: Migrate beacon to ouroboros-consensus-tools
https://github.com/input-output-hk/ouroboros-consensus-tools

### PR DESCRIPTION
**Pull request by dnadales**
Closes https://github.com/input-output-hk/ouroboros-consensus/issues/290

### CODE REVIEW

File: beacon/beacon.cabal [hunk 1]
```diff
@@ -0,0 +1,46 @@
+cabal-version:   3.0
+name:            beacon
+version:         0.1.0.0
+synopsis:
+  Benchmarking, exploration, and analisys of Consensus.
```
Comment (amesgen) [inline beacon/beacon.cabal]:
```suggestion
  Benchmarking, exploration, and analysis of Consensus.
```

File: cabal.project [hunk 2]
```diff
@@ -0,0 +1,18 @@
+repository cardano-haskell-packages
```
Comment (amesgen) [inline cabal.project:1]: This is just here to be future-proof in case we need sth from CHaP?

File: beacon/README.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+# Benchmarks, exploration, and analysis of Consensus
```
Comment (amesgen) [inline beacon/README.md:1]: Maybe at least link to our repo?

File: beacon/beacon.cabal [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+category:        Network
+build-type:      Simple
+extra-doc-files:
+  CHANGELOG.md
+  README.md
+
+executable beacon
+  hs-source-dirs: app
+  main-is:        beacon.hs
+  default-language: Haskell2010
```
Comment (amesgen) [inline beacon/beacon.cabal]: Given that this is a dev-only tool, we don't have to support GHC 8.10, so we could use GHC2021.
```suggestion
  default-language: GHC2021
```

**Review by amesgen — APPROVED** [review]
Thanks, LGTM :raised_hands: 

Do you already have local changes for a CI setup? If not, we could copy and slightly edit some stuff from ouroboros-consensus (happy to do that in a separate PR).

**Comment by dnadales** [comment]
Thanks for checking! 

> Do you already have local changes for a CI setup? If not, we could copy and slightly edit some stuff from ouroboros-consensus (happy to do that in a separate PR).

Not yet. I could migrate the GHA part, but I might need your help for the `nix` stuff :)

File: cabal.project (hunk 2 above)
Comment (dnadales) [reply cabal.project:1]: Indeed.

File: beacon/README.md (hunk 3 above)
Comment (dnadales) [reply beacon/README.md:1]: I linked it in the main README. Should I add it here as well?

Comment (amesgen) [reply beacon/README.md:1]: Ah, I missed that; I think this should be enough then :+1:

**Pull request merged**